packages/client/src/derivedProperties/derivedPropertyDefinitionFactory.ts

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2025 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { DerivedProperty } from "@osdk/api";
+import type { DerivedPropertyDefinition } from "@osdk/foundry.ontologies";
+import invariant from "tiny-invariant";
+
+/** @internal */
+export function derivedPropertyDefinitionFactory(
+  wireDefinition: DerivedPropertyDefinition,
+  definitionMap: Map<any, DerivedPropertyDefinition>,
+): DerivedProperty.NumericPropertyDefinition<any, any> & {
+  extractPart: DerivedProperty.DatetimePropertyDefinition<
+    any,
+    any
+  >["extractPart"];
+} {
+  const definition: DerivedProperty.NumericPropertyDefinition<any, any> & {
+    extractPart: DerivedProperty.DatetimePropertyDefinition<
+      any,
+      any
+    >["extractPart"];
+  } = {
+    abs() {
```

Comment:
We may want to come back through here later and do something clever with classes or prototypes but this is really just a future thought to myself more than anything else so you can pretty much disregard.